pkg/resource/deploy/step.go

```diff
@@ -1785,3 +1919,68 @@ func (s *DiffStep) Fail() {
 func (s *DiffStep) Skip() {
 	s.pcs.Reject(errors.New("skipped diff resource"))
 }
+
+// ViewStep isn't really a step like a normal step. It's really a virtual step for a view resource. The step itself
+// doesn't perform any operations against a provider, it's used to communicate the steps that were taken for the
+// view resource for display purposes.
+type ViewStep struct {
```

Comment:
An alternative approach would be folding this into the existing steps (e.g. `CreateStep`) and adding a few view-specific fields, including one that indicates it is a view step.